src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/parser/budget/SetExpenseLimitCommandParserTest.java

```diff
@@ -22,26 +25,21 @@ public void parse_emptyArg_throwsParseException() {
 
     @Test
     public void parse_invalidArgs_throwsParseException() {
-        assertParseFailure(parser, "a", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                SetExpenseLimitCommand.MESSAGE_USAGE));
-        assertParseFailure(parser, "-5", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                SetExpenseLimitCommand.MESSAGE_USAGE));
-        assertParseFailure(parser, "$", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                SetExpenseLimitCommand.MESSAGE_USAGE));
-        assertParseFailure(parser, "123.4", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                SetExpenseLimitCommand.MESSAGE_USAGE));
-        assertParseFailure(parser, "567.890", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                SetExpenseLimitCommand.MESSAGE_USAGE));
-        assertParseFailure(parser, "2 hello", String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                SetExpenseLimitCommand.MESSAGE_USAGE));
+        // no whitespace in front
+        assertParseFailure(parser, PREFIX_AMOUNT + VALID_AMOUNT_INTERNSHIP,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, SetExpenseLimitCommand.MESSAGE_USAGE));
+        // preamble before amount prefix
+        assertParseFailure(parser, "hello" + AMOUNT_DESC_INTERNSHIP,
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, SetExpenseLimitCommand.MESSAGE_USAGE));
     }
 
     @Test
     public void parse_validArgs_returnsSetExpenseLimitCommand() {
-        assertParseSuccess(parser, "500", new SetExpenseLimitCommand(new Amount("500")));
-        assertParseSuccess(parser, "314.15", new SetExpenseLimitCommand(new Amount("314.15")));
-        assertParseSuccess(parser, "$300", new SetExpenseLimitCommand(new Amount("$300")));
-        assertParseSuccess(parser, "$123.45", new SetExpenseLimitCommand(new Amount("$123.45")));
+        assertParseSuccess(parser, AMOUNT_DESC_INTERNSHIP,
+                new SetExpenseLimitCommand(new Amount(VALID_AMOUNT_INTERNSHIP)));
+        // whitespaces before/after keywords
+        assertParseSuccess(parser, "   " + AMOUNT_DESC_INTERNSHIP + "   ",
```

Comment:
```suggestion
        assertParseSuccess(parser, PREAMBLE_WHITESPACE + AMOUNT_DESC_INTERNSHIP + "   ",
```
You can use `PREAMBLE_WHITESPACE` instead of " ".